### PR TITLE
added PrometheusAvailabilityRatio alert

### PR DESCRIPTION
**Pull request by hervenicol**
Towards: https://github.com/giantswarm/giantswarm/issues/26770
Related: https://github.com/giantswarm/prometheus-rules/pull/729

This PR creates a new alert, `PrometheusAvailabilityRatio`, that pages during working hours when prometheus was up for less than 80% time in the past 10 hours.

Good to know: The alert is generated from MC prometheus, so if it's dead we have no visibility over it.

### Checklist

- [x] Update CHANGELOG.md
- [x] Add [Unit tests](https://github.com/giantswarm/prometheus-rules/#testing)
- [ ] Follow [Alert structure](https://github.com/giantswarm/prometheus-rules/#how-alerts-are-structured)
- [x] Consider [creating a dashboard](https://docs.giantswarm.io/getting-started/observability/grafana/custom-dashboards/) ([guidelines](https://intranet.giantswarm.io/docs/product/ux/guidelines/dashboards/)) (if it does not exist already) to help oncallers monitor the status of the issue.
- [ ] Request review from oncall area, as well as team (e.g: `oncall-kaas-cloud` GitHub group).

### CODE REVIEW

File: helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml [hunk 1]
```diff
@@ -10,6 +10,25 @@ spec:
   groups:
   - name: prometheus
     rules:
+    - alert: PrometheusAvailabilityRatio
```
Comment (QuentinBisson) [inline helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml:13]: I guess this is a temporary alert until we have an slo alert right? Can you add an ops-recipe link even if it's empty?

Comment (hervenicol) [reply helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml:13]: Well, I put the `prometheus-resource-limit-reached` opsrecipe because it explains quite well how to understand and fix all the reasons why prometheus would die.
That's better than an empty opsrecipe I guess?

Comment (hervenicol) [reply helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml:13]: Also, regarding SLOs:
I could have found a way to have an "SLO burn rate" alert. But I don't like them for a few reasons:
- they are quite generic and point to a generic opsrecipe
- the promql query is harder to understand

As I already had a nice query for this ratio in the availability dashboard, I thought I could as well use it.

Proper solution would be to use SLOTH, but it's not ready yet.

Comment (QuentinBisson) [reply helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml:13]: I agree with all your points, because even if we have an slo alert, each should have their different opsrecipe. I was thinking the expression behind your alert would be a good first step for our SLO

File: helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml [hunk 2]
```diff
@@ -23,7 +42,7 @@ spec:
         cancel_if_cluster_status_updating: "true"
         cancel_if_cluster_status_deleting: "true"
         cancel_if_cluster_has_no_workers: "true"
-        cancel_if_outside_working_hours: {{ include "workingHoursOnly" . }}
+        cancel_if_outside_working_hours: "false"
```
Comment (marieroque) [inline helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml:45]: Why this change ?

**Review by QuentinBisson — APPROVED** [review]
LGTM for a first attempt

File: helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml (hunk 2 above)
Comment (hervenicol) [reply helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml:45]: Short answer: The "workingHoursOnly" variable is very counter-intuitive and should not be used.

Longer answer:
It's defined here: https://github.com/giantswarm/prometheus-rules/blob/master/helm/prometheus-rules/templates/_helpers.tpl#L44-L50
It's set to `false` on most installations,
except for `openstack` and `capz` installations where it's set to `true`.
This kind of inconsistency is a nice recipe for hard-to-debug alerts :shrug:

**Review by TheoBrigitte — APPROVED** [review]

File: helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml (hunk 2 above)
Comment (marieroque) [reply helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml:45]: I agree.
This change affects the actual behaviour for capz right ?
I don't know why we have a specific case for capz BTW...

Comment (hervenicol) [reply helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml:45]: > This change affects the actual behaviour for capz right ?

Right.

> I don't know why we have a specific case for capz BTW...

Not sure there's a valid reason :shrug:
